pkg/key/key.go

```diff
@@ -3,9 +3,9 @@ package key
 // These consts are used by template.Params.Validate() to check that provided versions
 // fall within the known-good ranges.
 const (
-	CalicoVersionConstraint                       = ">= 3.10.0 < 3.16.0"
+	CalicoVersionConstraint                       = ">= 3.16.0 < 3.17.0"
 	EtcdVersionConstraint                         = ">= 3.4.0 < 3.5.0"
-	KubernetesVersionConstraint                   = ">= 1.16.0"
+	KubernetesVersionConstraint                   = ">= 1.17.0"
```

Comment:
1.16 is no longer supported so we can drop the logic for the old hyperkube binary extraction.